plugins/csp-wms-overlays/src/Plugin.hpp

```diff
@@ -0,0 +1,135 @@
+      std::vector<std::string> mWms; ///<	URLs of WMS servers.
+    };
+
+    std::map<std::string, Body> mBodies; ///< A list of bodies with their anchor names.
+  };
+
+  void init() override;
+  void deInit() override;
+
+  void update() override;
+
+ private:
+  void onLoad();
+
+  Settings::Body& getBodySettings(std::shared_ptr<TextureOverlayRenderer> const& wmsOverlay) const;
+
+  void setWMSServer(
+      std::shared_ptr<TextureOverlayRenderer> const& wmsOverlay, std::string const& name);
+  void resetWMSServer(std::shared_ptr<TextureOverlayRenderer> const& wmsOverlay);
+  void setWMSLayer(
+      std::shared_ptr<TextureOverlayRenderer> const& wmsOverlay, std::string const& name);
+  void resetWMSLayer(std::shared_ptr<TextureOverlayRenderer> const& wmsOverlay);
+  void setWMSStyle(
+      std::shared_ptr<TextureOverlayRenderer> const& wmsOverlay, std::string const& name);
+  void resetWMSStyle(std::shared_ptr<TextureOverlayRenderer> const& wmsOverlay);
+
+  /// Checks if the given wmsOverlay is currently active.
+  bool isActiveOverlay(std::shared_ptr<TextureOverlayRenderer> const& wmsOverlay);
+  /// Checks if the wmsOverlay with the given center is currently active.
+  bool isActiveOverlay(std::string const& center);
+
+  /// Adds the given layer and all of its sublayers to the layer dropdown.
+  /// Returns whether any layer's name matched the given activeLayer.
+  bool addLayerToSelect(std::shared_ptr<TextureOverlayRenderer> const& wmsOverlay,
+      WebMapLayer const& layer, std::string const& activeLayer, int const& depth = 0);
+
+  /// Move the observer so that the given bounds are visible.
+  void goToBounds(Bounds const& bounds);
+  /// Calculate the map scale for the given bounds and texture resolution. If it is outside the
+  /// appropriate range specified in the layer capabilities, a warning will be displayed.
+  void checkScale(Bounds const& bounds, WebMapLayer const& layer, int const& maxTextureSize);
+
+  std::shared_ptr<Settings> mPluginSettings = std::make_shared<Settings>();
+  std::map<std::string, std::shared_ptr<TextureOverlayRenderer>> mWMSOverlays;
+  std::map<std::string, std::vector<WebMapService>>              mWms;
+
+  std::shared_ptr<TextureOverlayRenderer> mActiveOverlay;
+  std::map<std::string, std::optional<WebMapService>>
+      mActiveServers; ///< The currently active WebMapService for each center name.
+  std::map<std::string, std::optional<WebMapLayer>>
+      mActiveLayers; ///< The currently active WebMapLayer for each center name.
+
+  bool mNoMovement;                ///< True when the observer is not moving.
+  bool mNoMovementRequestedUpdate; ///< True when the active overlay was requested to update its
+                                   ///< bounds because the observer is not moving.
+  std::chrono::time_point<std::chrono::high_resolution_clock>
+      mNoMovementSince; ///< Time at which the observer stopped moving.
```

Comment:
I would move the comments above the corresponding lines. This is more clang-format-friendly.